vllm/v1/core/scheduler.py

```diff
@@ -171,6 +170,16 @@ def schedule(self) -> "SchedulerOutput":
                     self.encoder_cache_manager.allocate(request, i)
                 encoder_budget = new_encoder_budget
 
+        #  Record the LoRAs in scheduled_running_reqs
+        requested_loras: set[int] = set()
+        if self.lora_config:
+            requested_loras =  \
+                set(req.lora_request.lora_int_id \
+                        for req in scheduled_running_reqs \
+                            if req.lora_request and \
+                                req.lora_request.lora_int_id > 0)
+            assert len(requested_loras) <= self.lora_config.max_loras
```

Comment:
Can we cache this state and incrementally update it whenever new request joins or finishes?

Reply:
I explored this a bit. Tracking the additions and deletions to the running queue in the current code is hard. The updates happen in more than one place (with new requests, finish requests and requests moving between running to preempted state and back). one way is to replace the append/remove/pop with  
```
self.running.<operation>()
if lora_config: 
    update_active_loras()
```
A better way is to subclass List and after any Create, Update, Delete operation we can update the active LoRAs. This is a considerable change. I believe we can do this after some profiling to see how bad this code is. 
For the moment, I think this localized update is nicer as it doesn't introduce a bunch of `if self.lora_config`s . 

Is there a better way I am missing ?